playlist.c

```diff
@@ -1870,7 +2002,13 @@ static JSON_Parser_HandlerResult JSONEndArrayHandler(JSON_Parser parser)
    else if (pCtx->object_depth == 2)
    {
       if (pCtx->in_subsystem_roms && string_is_equal(pCtx->current_items_string, "subsystem_roms") && pCtx->array_depth == 1)
+      {
          pCtx->in_subsystem_roms = false;
+      }
+      else if (pCtx->in_subsystem_roms && string_is_equal(pCtx->current_items_string, "subsystem_roms_relative_paths") && pCtx->array_depth == 1)
```

Comment:
`pCtx->in_subsystem_roms` should be `pCtx->in_subsystem_roms_relative_paths` here

Also, you can move the `pCtx->array_depth == 1` check to a higher level